roles/manage_ec2_instance/tasks/ec2_instance_create_operations.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Verify that an instance with same name does not exist
+  block:
+    - name: Get instance info with provided name
+      amazon.aws.ec2_instance_info:
+        filters:
+          tag:Name: "{{ manage_ec2_instance_instance_name }}"
+          instance-state-name: ["pending", "running", "stopping", "stopped"]
+      register: ec2_info_result
+
+    - name: Print warning and exit if instance exists
+      ansible.builtin.fail:
+        msg: "Instance with name {{ manage_ec2_instance_instance_name }} already exists in {{ aws_region }}. Please provide a different name to avoid updating the existing instance."
+      when: ec2_info_result.instances | length > 0
+
+- name: Create a key pair if required
+  when: manage_ec2_instance_key_name is defined and manage_ec2_instance_key_name | length > 0
+  block:
+    - name: Get key pair info
+      amazon.aws.ec2_key_info:
+        names:
+          - "{{ manage_ec2_instance_key_name }}"
+      register: key_info_result
+
+    - name: Create new key pair
+      amazon.aws.ec2_key:
+        name: "{{ manage_ec2_instance_key_name }}"
+        state: present
+      when: key_info_result.keypairs | length == 0
+      register: new_key_pair_result
+
+- name: Create EC2 instance with provided configuration
+  amazon.aws.ec2_instance:
+    state: running
+    name: "{{ manage_ec2_instance_instance_name }}"
+    instance_type: "{{ manage_ec2_instance_instance_type }}"
+    image_id: "{{ manage_ec2_instance_ami_id }}"
+    key_name: "{{ manage_ec2_instance_key_name | default(omit) }}"
+    security_groups: "{{ manage_ec2_instance_associate_security_groups | default(omit, true) }}"
+    vpc_subnet_id: "{{ manage_ec2_instance_vpc_subnet_id | default(omit) }}"
+    tags: "{{ manage_ec2_instance_tags | default(omit) }}"
+    wait: "{{ manage_ec2_instance_wait_for_state }}"
+  register: ec2_instance
+
+- name: Allocate and associate Elastic IP if enabled
+  when: manage_ec2_instance_associate_eip is true
+  amazon.aws.ec2_eip:
+    device_id: "{{ ec2_instance.instance_ids[0] }}"
+    state: present
+    release_on_disassociation: true
+  register: instance_eip
+
+- name: Get EC2 instance info
+  amazon.aws.ec2_instance_info:
+    instance_ids: "{{ ec2_instance.instance_ids[0] }}"
+  register: _ec2_instance
+
+- name: Output details of the created EC2 instance
+  ansible.builtin.debug:
+    msg:
+      - "EC2 instance {{ ec2_instance.instance_ids[0] }} created successfully"
+      - "Instance details: {{ _ec2_instance.instances[0] }}"
+
+- name: Output private key if a new keypair was created
+  when: new_key_pair_result.key is defined
+  ansible.builtin.debug:
+    msg: "A new key pair was created for ssh access to the instance. Please save this private key for reference, it will not be accessible again: {{ new_key_pair_result.key.private_key }}"
```

Comment:
Arguably we would also need to save the instance ip address. Since the private key is available via `new_key_pair_result.key.private_key` just like the ec2 instance info is available via `_ec2_instance.instances[0]`, I think it's fine to leave as is for now.

Reply:
If we leave it as it is, the role is unusable as there is no means to connect to the instance created (no private key file and IP address), the only use case is spawning an instance with a key pair and IP

Reply:
The two vars I mentioned above are available to any playbook that uses this role. A user can reference them in the same way that we suggest they reference `ec2_networking_resources_subnet_result.subnet.id` when combining this role with the ec2_networking resources role. I can update the var names and the README to make that more clear, but we don't need to create new variables.

Reply:
Updated, I think it should be more clear now.